pkg/services/rpcsrv/server.go

```diff
@@ -3202,3 +3204,102 @@ func (s *Server) getRawNotaryTransaction(reqParams params.Params) (any, *neorpc.
 	}
 	return tx.Bytes(), nil
 }
+
+// getBlockNotifications returns notifications from a specific block with optional filtering.
+func (s *Server) getBlockNotifications(reqParams params.Params) (any, *neorpc.Error) {
+	param := reqParams.Value(0)
+	hash, respErr := s.blockHashFromParam(param)
+	if respErr != nil {
+		return nil, respErr
+	}
+
+	block, err := s.chain.GetBlock(hash)
```

Comment:
I think we need some `GetTrimmedBlock` or alike, this one does extract all transactions and we don't need them (only hash is needed for `GetAppExecResults()`). But it's an optimization in its essence, current code will work fine.